sleap_io/io/nwb.py

```diff
@@ -0,0 +1,282 @@
+) -> ProcessingModule:
+    """Auxiliary function to create a processing module.
+    
+    Checks for the processing module existence and creates if not available.
+
+    Args:
+        processing_module_name (str): The name of the processing module.
+        nwbfile (NWBFile): The nwbfile to attach the processing module to.
+
+    Returns:
+        ProcessingModule: An nwb processing module with the desired name.
+    """
+    if processing_module_name in nwbfile.processing:
+        nwb_processing_module = nwbfile.processing[processing_module_name]
+    else:
+        nwb_processing_module = nwbfile.create_processing_module(
+            name=processing_module_name, description=f"Processed SLEAP data"
+        )
+
+    return nwb_processing_module
+
+
+def build_pose_estimation_container_for_track(
+    labels_data_df: pd.DataFrame, labels: Labels, track_name: str, video: Video
+) -> PoseEstimation:
+    """Creates a PoseEstimation container for a track.
+
+    Args:
+        labels_data_df (pd.DataFrame): A pandas object with the data corresponding
+        to the predicted instances associated to this labels object.
+        labels (Labels): A general labels object
+        track_name (str): The name of the track in labels.tracks
+        video (Video): The video to which data belongs to
+
+    Returns:
+        PoseEstimation: A PoseEstimation multicontainer where the time series
+        of all the node trajectories in the track are stored. One time series per 
+        node.
+    """
+    video_path = Path(video.filename)
+
+    all_track_skeletons = (
+        labels_data_df[video.filename, track_name]
+        .columns.get_level_values("skeleton_name")
+        .unique()
+    )
+    # Assuming only one skeleton per track
+    skeleton_name = all_track_skeletons[0]
+    skeleton = next(
+        skeleton for skeleton in labels.skeletons if skeleton.name == skeleton_name
+    )
+
+    track_data_df = labels_data_df[
+        video.filename,
+        track_name,
+        skeleton.name,
+    ]
+
+    pose_estimation_series_list = build_track_pose_estimation_list(track_data_df)
+
+    # Combine each node's PoseEstimationSeries to create a PoseEstimation container
+    container_description = (
+        f"Estimated positions of {skeleton.name} in video {video_path.name} "
+        f"using SLEAP."
+    )
+    pose_estimation_container = PoseEstimation(
+        name=f"track={track_name}",
+        pose_estimation_series=pose_estimation_series_list,
+        description=container_description,
+        original_videos=[f"{video.filename}"],
+        labeled_videos=[f"{video.filename}"],
+        source_software="SLEAP",
+        nodes=skeleton.node_names,
+        edges=np.array(skeleton.edge_inds).astype("uint64"),
+        # dimensions=np.array([[video.backend.height, video.backend.width]]),
+        # scorer=str(labels.provenance),
+        # source_software_version=f"{sleap.__version__}",
+        # To-discuss this in PR.
```

Comment:
Is there an example data of yours that has provenance so I can see how this could be written?
I also think that appending the software version directly (that is, the assumption that the writing is taking place in the same computer that the predictions were done) is probably not a good idea. Maybe that's something that should be on provenance as well?

Reply:
Right, these are usually stored in the provenance metadata, but I think the dataset we're currently using as a fixture here might be a bit too old from before we saved that together with the predictions. We can re-generate it using a newer SLEAP version for this PR.